kolibri/core/assets/src/core-actions.js

```diff
@@ -349,6 +443,85 @@ function stopTrackingProgress(store, Kolibri) {
   updateTimeSpent(store, Kolibri, true);
 }
 
+function saveMasteryLog(store, Kolibri) {
+  const masteryLogModel = Kolibri.resources.MasteryLog.getModel(store.core.logging.mastery.id);
+  masteryLogModel.save(_masteryLogModel(store)).then((newMasteryLog) => {
```

Comment:
@rtibbles gave us a new tool called the `ConditionalPromise` that adds a chainable `.only` method to promises. This helps us choose to avoid calling the `.then` handler when it would be inappropriate due to a race condition.

A typical use-case is if an API request returns but the user has already navigated to a new page - we need to avoid clobbering the page state in the callback. Check out `actions.js` for examples of it in use.